doc/changelog.md

```diff
@@ -1,3 +1,55 @@
+# Release 0.4.1
+
+<h3>Improvements</h3>
+
+* Catalyst wheels are now packaged with OpenMP and ZStd.
+  [(#457)](https://github.com/PennyLaneAI/catalyst/pull/457)
+
+<h3>Bug fixes</h3>
```

Comment:
Looking into the main [changelog](https://github.com/PennyLaneAI/catalyst/blob/main/doc/changelog.md), references to the following PRs are missing,
- https://github.com/PennyLaneAI/catalyst/pull/409
- https://github.com/PennyLaneAI/catalyst/pull/446
- https://github.com/PennyLaneAI/catalyst/pull/473

Reply:
added them as code suggestions to this PR